comp/snmpscan/impl/snmpscan_test.go

```diff
@@ -0,0 +1,36 @@
+//go:build test
+
+package snmpscanimpl
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/comp/aggregator/demultiplexer/demultiplexerimpl"
+	"github.com/DataDog/datadog-agent/comp/core"
+	snmpscan "github.com/DataDog/datadog-agent/comp/snmpscan/def"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/gosnmp/gosnmp"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/fx"
+)
+
+func TestSnmpScanComp(t *testing.T) {
+	deps := fxutil.Test[snmpscan.Requires](
+		t,
+		fx.Supply(core.BundleParams{}),
+		demultiplexerimpl.MockModule(),
+		core.MockBundle(),
+	)
```

Comment:
Since we are transition to having no fx reference in the `impl` folder we are going through a bit of a migration in which writing test is not straight forward as the ASM would want, sorry about that. 

You could try doing something like: 

```go
import	logmock "github.com/DataDog/datadog-agent/comp/core/log/mock"

type deps struct {
	fx.In
	Demultiplexer demultiplexer.Mock
}


lc := compdef.NewTestLifecycle(t)
logComponent := logmock.New(t)
testDeps := fxutil.Test[deps](t, demultiplexerimpl.MockModule(), defaultforwarder.MockModule(), core.MockBundle())

requires := Requires{
	Logger: logComponent,
	Demultiplexer: testDeps.Demultiplexer
}
```

let me know if you have any doubt 😄

Reply:
Oh indeed this works, I though importing fx.In in the test would also fail but looks like it's fine because it's a test?
Seems to be working now, thanks!